gapic/src/main/com/google/gapid/util/UpdateWatcher.java

```diff
@@ -66,14 +66,16 @@ private void scheduleCheck() {
   private void doCheck() {
     SettingsProto.Preferences.Builder prefs = settings.writePreferences();
     if (prefs.getCheckForUpdates()) {
-      ListenableFuture<Release> future = client.checkForUpdates(prefs.getIncludeDevReleases());
+      ListenableFuture<Service.Releases> future =
+          client.checkForUpdates(prefs.getIncludeDevReleases());
       prefs.setUpdateAvailable(false);
       try {
-        Release release = future.get();
-        if (release != null) {
+        Service.Releases releases = future.get();
```

Comment:
Not directly related to this PR's intent but a question on our previous implementation around this part: It looks that we didn't use a RPC.listen mechanism here, so I'm curious would we actually get the future object's result by fetching this quickly here?

Reply:
`RPC.listen` is asynchronous. Here, since we are in our own thread, we are fine to be synchronous.